summarization/stamp_page_picking/interesting_sequence_picker.py

```diff
@@ -0,0 +1,233 @@
+            = preprocessor.get_cover_objects_for_stamp_pages()
+
+    def _pick_cover_at_index(self, index):
+        # update the picked_cover
+        for i in range(self.cover_size):
+            self.picked_cover[i] \
+                = self.picked_cover[i] | self.stamp_page_covers[index].cover[i]
+
+    def _get_stamp_page_type(self, stamp_page_index):
+        # assign a number to each stamp page type
+        stamp_page = self.stamp_pages[stamp_page_index]
+        if stamp_page.is_embedded_content:
+            return 1
+        elif stamp_page.media_index != -1 and stamp_page.sentence_index != -1:
+            return 2
+        elif stamp_page.media_index != -1:
+            return 3
+        elif stamp_page.sentence_index != -1:
+            return 4
+
+    def _get_content_change_score(self, from_index, to_index):
+        # higher change in content type is better
+        return abs(
+            self._get_stamp_page_type(from_index),
+            self._get_stamp_page_type(to_index)
+        )
+
+    def _get_unpicked_weights_to_cost_ratio(self, index):
+        total_weight = 0
+        for i in range(self.cover_size):
+            if self.picked_cover[i] == 0 \
+                    and self.stamp_page_covers[index].cover[i] == 1:
+                total_weight += 1
+        return total_weight / self.stamp_page_covers[index].cost
+
+    def _compute_weighted_score(self, score_list):
+        return sum(
+            [a * b for a, b in zip(score_list, self.individual_score_weights)]
+        )
+
+    def _get_approximate_index_for_stamp_page(self, stamp_page_index):
+        '''Fetches the approximate index so it can be used
+        for computing its distance to the sweepeing index
+        '''
+        index_sum = 0
+        valid_index_count = 0
+        stamp_page = self.stamp_pages[stamp_page_index]
+
+        if stamp_page.media_index != -1:
+            index_sum += stamp_page.media_index
+            valid_index_count += 1
+
+        if stamp_page.sentence_index != -1:
+            index_sum += stamp_page.sentence_index
+            valid_index_count += 1
+
+        return index_sum / valid_index_count
+
+    def _get_distance_from_sweeping_index(self, stamp_page_index):
+        approximate_index = self._get_approximate_index_for_stamp_page(
+            stamp_page_index)
+        return abs(self.sweeping_index - approximate_index)
+
+    def _interestingess_metric(self, stamp_page_index):
+        '''
+        calculates the interesting-ness metric based
+        on the last picked stamp page
+        '''
+        from_stamp_page_index = self.last_picked_stamp_page_index
+        to_stamp_page_index = stamp_page_index
+
+        content_change_score \
+            = self._get_content_change_score(
+                from_stamp_page_index,
+                to_stamp_page_index
+            )
+
+        sentiment_change_score \
+            = 0  # will be amended once sentiment detection PR is added
```

Comment:
Consider splitting individual score's calculation into different classes. Calculating all 4 here could make this class unnecessarily big.

Reply:
I see. But the score calculations require a lot of variables that are in the scope of this class - we might need to again pass all of them to another class/object . WDYT?

Reply:
That should still be better than having everything in one class IMO. Creating interfaces might also make sense here.

Reply:
Updated accordingly!